libs/x-content/src/main/java/org/elasticsearch/common/xcontent/ObjectParser.java

```diff
@@ -235,11 +236,32 @@ public Value parse(XContentParser parser, Value value, Context context) throws I
                     unknownFieldParser.acceptUnknownField(name, currentFieldName, currentPosition, parser, value, context);
                 } else {
                     fieldParser.assertSupports(name, parser, currentFieldName);
+
+                    // Check to see if this field is a required field, if it is we can
+                    // remove the entry as the requirement is satisfied
+                    Iterator<String[]> iter = requiredFieldSets.iterator();
+                    while (iter.hasNext()) {
+                        String[] requriedFields = iter.next();
+                        for (String field : requriedFields) {
+                            if (field.equals(currentFieldName)) {
+                                iter.remove();
```

Comment:
Isn't this removing the value from the ObjectParser's instance, which is static?

Reply:
Oof, you're right.  I was just looking at the class itself (which is not static) but everyone declares ObjectParser as static.  I'll fiddle to get rid of the mutability